governance-server/src/aeternity.js

```diff
@@ -181,53 +188,55 @@ module.exports = class Aeternity {
         return this.cache.getOrSet(["height"], () => this.client.height(), this.cache.shortCacheTime);
     };
 
-    transactionEvent = async (hash) => {
-        return this.cache.getOrSet(["transactionEvent", hash], async () => {
+    transactionEvent = async ({hash, tx}) => {
+        return this.cache.getOrSet(["transactionEvent", hash || tx.hash], async () => {
             process.stdout.write(",");
-            const tx = await this.client.getTxInfo(hash);
-            if (tx.log.length === 1) {
+            const {height, nonce, log} = hash
+              ? await this.client.getTxInfo(hash).then(res => ({height: res.height, nonce: res.callerNonce, log: res.log}))
```

Comment:
why don't we always save the full TX under the hash?

Reply:
when we only have the hash,  which is the case if we iterate the blockchain manually to find delegation event, we need to first get the full transaction to have its event log. With the new MDW we receive the full tx directly, so we can skip fetching the tx from chain.